server/src/test/java/org/elasticsearch/index/mapper/blockloader/TextFieldBlockLoaderTests.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.mapper.blockloader;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.index.mapper.BlockLoaderTestCase;
+import org.elasticsearch.logsdb.datageneration.FieldType;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class TextFieldBlockLoaderTests extends BlockLoaderTestCase {
+    public TextFieldBlockLoaderTests(Params params) {
+        super(FieldType.TEXT.toString(), params);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected Object expected(Map<String, Object> fieldMapping, Object value, TestContext testContext) {
+        if (fieldMapping.getOrDefault("store", false).equals(true)) {
+            return valuesInSourceOrder(value);
+        }
+
+        var fields = (Map<String, Object>) fieldMapping.get("fields");
+        if (fields != null) {
+            var keywordMultiFieldMapping = (Map<String, Object>) fields.get("kwd");
+            boolean docValues = hasDocValues(keywordMultiFieldMapping, true);
+            boolean index = keywordMultiFieldMapping.getOrDefault("index", true).equals(true);
+            boolean store = keywordMultiFieldMapping.getOrDefault("store", false).equals(true);
+            Object ignoreAbove = keywordMultiFieldMapping.get("ignore_above");
+
+            // See TextFieldMapper.SyntheticSourceHelper#usingSyntheticSourceDelegate
+            // and TextFieldMapper#canUseSyntheticSourceDelegateForLoading().
+            boolean usingSyntheticSourceDelegate = docValues || store;
+            boolean canUseSyntheticSourceDelegateForLoading = usingSyntheticSourceDelegate && ignoreAbove == null && (index || store);
```

Comment:
I don't really understand why do we require multi field to be indexed here (the actual impl is in `TextFieldMapper`).

Reply:
Yes, that is strange (only doc values / stored should matter). What failure occurs when we don't check for `index` here?

Reply:
I think you'll get different results sometimes, i can't remember now.